classes/sync.php

```diff
@@ -107,22 +107,24 @@ public function sync(progress_trace $trace) {
                 if (!$rs->EOF) {
                     while ($fields = $rs->FetchRow()) {
                         $count = array_pop($fields);
-                        if ($count > $minrecords) {
+                        if ($count >= $minrecords) {
                             $hasenoughrecords = true;
                         }
                     }
                 }
             }
         }
         if (!$hasenoughrecords) {
-            mtrace("Failed to sync because the external db returned $count records and the minimum required is $minrecords");
+            $message = "Failed to sync because the external db returned $count records and the minimum required is $minrecords";
+            $this->email_admins($message);
+            mtrace($message);
             $trace->finished();
             return 1;
         }
 
         // Get list of current cohorts indexed by idnumber.
         $cohortrecords = $DB->get_records('cohort', array('component' => 'tool_cohortdatabase'),
-            '', 'idnumber, id, name, description, contextid');
+            '', 'idnumber, id, name, description, contextid, descriptionformat, visible, component, timecreated, timemodified, theme');
```

Comment:
Theme field was added in 3.5, so there should be a version bump and require change as the plugin supports 3.4

Reply:
good catch.